src/fal/cli/selectors.py

```diff
@@ -61,21 +61,32 @@ def create_plan_from_graph(cls, parsed, nodeGraph: NodeGraph, fal_dbt: FalDbt):
 def _filter_node_ids(
     unique_ids: List[str],
     fal_dbt: FalDbt,
-    selected_nodes: List[str],
+    selectors: List[str],
     nodeGraph: NodeGraph,
 ) -> List[str]:
     """Filter list of unique_ids according to a selector."""
     output = []
 
-    for selector in selected_nodes:
-        selector_plans = map(
-            lambda sub_selector: SelectorPlan(sub_selector, unique_ids, fal_dbt),
-            SET_INTERSECTION.separate(selector),
-        )
-        selector_outputs = map(
-            lambda plan: set(plan.execute(nodeGraph)), selector_plans
-        )
-        output.extend(set.intersection(*selector_outputs))
+    for selector in selectors:
+        selector_plan_groups = [
+            [
+                SelectorPlan(sub_selector, unique_ids, fal_dbt)
+                for sub_selector in union.split(OP_SET_INTERSECTION)
+            ]
+            for union in selector.split(OP_SET_UNION)
+        ]
+        selector_outputs = [
+            set.intersection(
+                *[
+                    set(plan.execute(nodeGraph))
+                    for plan in selector_plan_group
+                    # Initially group each set by intersection, and then in
+                    # the next step we'll take a union of all these intersections.
+                ]
+            )
+            for selector_plan_group in selector_plan_groups
+        ]
+        output.extend(set.union(*selector_outputs))
```

Comment:
Can we add a test like this:

```
# .../integration_tests/projects/008_pure_python_models
❯ dbt ls -s "model_a model_b" , "model_b model_c"
fal_008.model_a
fal_008.model_b
fal_008.fal.staging.model_c
```

I have my doubts because we are processing each `selector` separately. Notice there is a space around `,`

Reply:
Oh, this is an interesting case. I would thought this to be an explicit syntax error, if we'd assume " " as an operator. But apparently it is supported by DBT. I'll check it out, thanks!

Reply:
I think this can be handled by doing `" ".join(select)` and just passing the single string to the selectors code

Reply:
As a side-note, I think this is the sort of thing that might change in future versions of dbt and we will have to change with it. How should we keep track of this?

Reply:
Maybe a better way would be using the same algorithm from DBT? As far as I understand, we can't explicitly depend on DBT (since `fal` does not depend on `dbt-core`) but we can still carry over the whole function (it seems pretty-isolated):
https://github.com/dbt-labs/dbt-core/blob/32e1924c3bd67e2e3f8d4bc0ec336682e910cd80/core/dbt/graph/cli.py#L27-L53

We just need to change the representation of `SelectionCriteria` (and stuff like it) to more basic data types.

Reply:
Oh yes, that `parse_union` function looks pretty good.

Reply:
True, if we don't depend on dbt-core directly we need to still keep it up to date manually. Is there any links to the upcoming changes that you know? Do they plan to make some backwards incompatible changes on the parsing of unions/intersections?

Reply:
Not as far as I know. This is just me being a little paranoid.

Reply:
We can use `from dbt.grapg.cli import parse_union`, it will import the correct one based on the installed dbt version. dbt-core is a peer-dependency for us, I think we don't state it anywhere though.

Reply:
Thanks @chamini2 for the information. During standup, we have decided to just copy this over since it is a relatively simple code and is unlikely to change in the short term.